core/src/apps/nem/transfer/serialize.py

```diff
@@ -78,32 +82,37 @@ def serialize_importance_transfer(
     return w
 
 
-def get_transfer_payload(transfer: NEMTransfer, node) -> [bytes, bool]:
+def get_transfer_payload(
+    transfer: NEMTransfer, node: bip32.HDNode
+) -> tuple[bytes, bool]:
```

Comment:
it turns out that `_encrypt` will fail if `payload == b""`

suggest the following changes

in protobuf, specify `optional bytes payload [default=""]` so that we know that payload is always bytes and don't need to mess with the `None` case

here, the code can be:
```
if transfer.public_key is not None:
    if not transfer.payload:
       raise ...
    encrypted_payload = _encrypt(...)
    return encrypted_payload, True
else:
    return transfer.payload, False
```

Reply:
Nice suggestion, it is even simplifying the function. Done in 8fcd268